readers/elasticsearch/src/main/java/com/criteo/hadoop/garmadon/elasticsearch/ElasticSearchReader.java

```diff
@@ -105,6 +109,11 @@
     void writeToES(GarmadonMessage msg) {
         String msgType = GarmadonSerialization.getTypeName(msg.getType());
         long timestampMillis = msg.getTimestamp();
+
+        if (GarmadonSerialization.TypeMarker.APPLICATION_EVENT == msg.getType()) elasticSearchCacheManager.addAppEventInCache(msg);
```

Comment:
I propose to use a single method -> esCacheManager.cacheEnrichableData(garmadonMsg) and this method has the intelligence to know what event type has what data that can be used in enrichment.

The EsCacheManager itself would be a proxy to several instances of msg specific caches